app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,70 @@
+class CoursesController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+    @courses = Course.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    @course = Course.find_by(id: params[:id])
```

Comment:
```
 @user = User.find_by(id: params[:id])
 @course = Course.find_by(id: params[:id])
```
why do you find user, and course with the same `params[:id]` value??

Reply:
I think since I already define different ID when the course created/register, the ID here will be different. I tried to change it but I got errors and problems to link the user to the gravatar.